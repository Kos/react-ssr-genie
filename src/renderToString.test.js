/* eslint-env jest */
const { createElement: h } = require('react');
const { default: styled } = require('styled-components');
const renderToString = require('./renderToString');
/* workaround for https://github.com/styled-components/styled-components/issues/893 */
const { default: StyleSheet } = require('styled-components/lib/models/StyleSheet');
StyleSheet.reset(true); // reset to use server stylesheet

function ExampleComponent({ text = 'Hello', isBold = false }) {
  return h('div', { style: isBold ? { fontWeight: 'bold' } : undefined }, text);
}

const Box = styled.div`
  background: ${props => (props.emphasis ? 'hotpink' : 'pink')};
`;

describe('renderToString', () => {
  it('should render with default props', () => {
    expect(renderToString(ExampleComponent, {})).toMatchSnapshot();
  });

  it('should render with interesting props', () => {
    expect(renderToString(ExampleComponent, { text: 'changes', isBold: true })).toMatchSnapshot();
  });

  it('should render a styled component', () => {
    expect(renderToString(Box, {})).toMatchSnapshot();
  });

  it('should render a styled component with a prop', () => {
    expect(renderToString(Box, { emphasis: true })).toMatchSnapshot();
  });
});
