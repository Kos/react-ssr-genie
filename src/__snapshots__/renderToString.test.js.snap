// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderToString should render a styled component 1`] = `
"<style type=\\"text/css\\" data-styled-components=\\"hdntL\\" data-styled-components-is-local=\\"true\\">/* sc-component-id: sc-bdVaJa */
.sc-bdVaJa {}

.hdntL{background:pink;}
</style>
<div class=\\"sc-bdVaJa hdntL\\"></div>"
`;

exports[`renderToString should render a styled component with a prop 1`] = `
"<style type=\\"text/css\\" data-styled-components=\\"cfPElA\\" data-styled-components-is-local=\\"true\\">/* sc-component-id: sc-bdVaJa */
.sc-bdVaJa {}

.cfPElA{background:hotpink;}
</style>
<div class=\\"sc-bdVaJa cfPElA\\"></div>"
`;

exports[`renderToString should render with default props 1`] = `
"<style type=\\"text/css\\" data-styled-components=\\"\\" data-styled-components-is-local=\\"true\\"></style>
<div>Hello</div>"
`;

exports[`renderToString should render with interesting props 1`] = `
"<style type=\\"text/css\\" data-styled-components=\\"\\" data-styled-components-is-local=\\"true\\"></style>
<div style=\\"font-weight:bold\\">changes</div>"
`;
